metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala

```diff
@@ -0,0 +1,79 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.parsing.Trees
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class ConvertToNamedArguments(trees: Trees) extends CodeAction {
+
+  import ConvertToNamedArguments._
+  override val kind: String = l.CodeActionKind.RefactorRewrite
+
+  def firstApplyWithUnnamedArgs(
+      term: Option[Tree]
+  ): Option[ApplyTermWithArgIndices] = {
+    term match {
+      case Some(apply: Term.Apply) =>
+        val argIndices = apply.args.zipWithIndex.collect {
+          case (arg, index)
+              if !arg.isInstanceOf[Term.Assign] && !arg
+                .isInstanceOf[Term.Block] =>
+            index
+        }
+        if (argIndices.isEmpty) firstApplyWithUnnamedArgs(apply.parent)
+        else Some(ApplyTermWithArgIndices(apply, argIndices))
+      case Some(t) => firstApplyWithUnnamedArgs(t.parent)
+      case _ => None
+    }
+  }
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = {
+
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+
+    val maybeApply = for {
+      term <- trees.findLastEnclosingAt[Term.Apply](path, range.getStart())
```

Comment:
```suggestion
      term <- trees.findLastEnclosingAt[Term.Apply](path, range.getStart())
      if !term.fun.encloses(range)
```
So that we don't get the code action when at for example `F<<>>uture.succesfull(1)`